src/Program.cs

```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
                 if (File.Exists(inputPath))
                 {
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    HtmlProcessor.ProcessFile(inputPath, outputPath);
+
+                    // Check file extension to decide whether to use MdProcessor or HtmlProcessor
+                    string fileExtension = Path.GetExtension(inputPath);
+                    if (fileExtension == ".md")
+                    {
+                        MdProcessor.ProcessFile(inputPath, outputPath);
```

Comment:
The class `HtmlProcessor` should be only class handling this. But there should be two separate functions for handling `.md` versus `.txt`. The `ProcessFile` function should be called from `HtmlProcessor`. This function however should be modified to contain logic for handling `.txt` versus `.md`. `ProcessFile` is the decider of which processor function to call.